torch/lib/c10d/ProcessGroupGloo.hpp

```diff
@@ -68,20 +68,31 @@ class ProcessGroupGloo : public ProcessGroup {
   //
   class AsyncWork : public ProcessGroup::Work {
    public:
+    AsyncWork() {
+      resultFuture = c10::make_intrusive<c10::ivalue::Future>(
+          c10::ListType::create(c10::TensorType::get()));
+    }
+
+    c10::intrusive_ptr<c10::ivalue::Future> getFuture() override {
```

Comment:
We should update our docs for `get_future` to reflect this now works for Glooo as well: https://github.com/pytorch/pytorch/blob/master/torch/csrc/distributed/c10d/init.cpp#L756 (except send/recv).